src/Symfony/Component/Security/Http/Tests/Firewall/ChannelListenerTest.php

```diff
@@ -153,4 +155,30 @@ public function testHandleWithSecuredRequestAndHttpChannel()
 
         $this->assertSame($response, $event->getResponse());
     }
+
+    public function testSupportsWithoutHeaders()
+    {
+        $request = $this->createMock(Request::class);
+        $request
+            ->expects($this->any())
+            ->method('isSecure')
+            ->willReturn(false)
+        ;
+        $request->headers = new HeaderBag();
+
+        $accessMap = $this->createMock(AccessMapInterface::class);
+        $accessMap
+            ->expects($this->any())
+            ->method('getPatterns')
+            ->with($this->equalTo($request))
+            ->willReturn([[], 'https'])
+        ;
+
+        $entryPoint = $this->createMock(AuthenticationEntryPointInterface::class);
+        $logger = $this->createMock(LoggerInterface::class);
+
+        $listener = new ChannelListener($accessMap, $entryPoint, $logger);
```

Comment:
The logger is optional.

```suggestion
        $listener = new ChannelListener($accessMap, $entryPoint);
```

Reply:
Actually, not for what I'm testing here, as that requires the logger to be set 😄 

https://github.com/symfony/symfony/blob/b45560c74620873ac04b26a2562ddee20e9a444a/src/Symfony/Component/Security/Http/Firewall/ChannelListener.php#L51

Reply:
Oh, I see. Let's pass a `NullLogger` instead then.

Reply:
Ok, it's updated.